internal/gatewayapi/securitypolicy.go

```diff
@@ -147,6 +148,17 @@ func (t *Translator) ProcessSecurityPolicies(securityPolicies []*egv1a1.Security
 					continue
 				}
 
+				if err := validation.ValidateSecurityPolicy(policy); err != nil {
+					status.SetTranslationErrorForPolicyAncestors(&policy.Status,
+						parentGateways,
+						t.GatewayControllerName,
+						policy.Generation,
+						fmt.Sprintf("Invalid SecurityPolicy: %v", err),
```

Comment:
we should probably use `status.Error2ConditionMsg(err)` here as the rest of the file